packages/consensus/src/validation/block.cairo

```diff
@@ -47,6 +50,50 @@ pub fn compute_and_validate_tx_data(
 
         let txid = double_sha256_byte_array(tx_bytes_legacy);
 
+        // Removes the outpoint hash of a transaction input if it was in the cache.
```

Comment:
It’s better to put this logic inside validate_transactions where we already have loops over inputs and outputs (to minimize number of loops).

Reply:
if i do that, then coinbase is excluded, i should handle it separately ?

Reply:
ah you mean only inputs loop right?

Reply:
well i suppose its not needed as coinbase tx is treated separately, and will never be cached as it cannot be spend in the same block and there will be a check for that